mne/decoding/transformer.py

```diff
@@ -245,6 +245,142 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(TransformerMixin):
+    """Vectorizer transforms different object data to fit into a scikit-learn
+       pipeline.
+
+    Parameters
+    ----------
+    info : instance of Info
+        The measurement info.
+
+    Attributes
+    ----------
+    n_channels : int
+        The number of channels.
+    n_times : int
+        The number of time points.
+
+    """
+    def __init__(self, info=None):
+        self.info = info
+        self.n_channels = None
+        self.n_times = None
+
+    def fit(self, data, y=None):
+        """Concatenate data from different channels into a single
+        feature vector.
+
+        Parameters
+        ----------
+        data : array, shape (n_samples, n_channels) or
+                      (n_samples, n_channels, n_freq) or
+                      (n_samples, n_channels, n_freq, n_time)
+        y : None | array, shape (n_samples,)
+            The label for each vector.
+            If None not used. Defaults to None.
+
+        Returns
+        -------
+        self : instance of Vectorizer
+            returns the modified instance
+        """
+        if not isinstance(data, np.ndarray):
+            raise ValueError("data should be of type ndarray (got %s)."
+                             % type(data))
+
+        if data.ndim == 2:
+            self.shape = 2
+
+        elif data.ndim == 3:
+            self.shape = 3
+            # For epochs or PyRiemann Covariance
+            self.n_samples, self.n_channels, self.n_times = data.shape
+
+        elif data.ndim == 4:
+            self.shape = 4
```

Comment:
if you have an attribute that is written or updated by calling .fit you should let it end with an underscore, hence, `.shape_`